evap/evaluation/auth.py

```diff
@@ -149,3 +152,44 @@ def check_user(user):
         from evap.rewards.tools import can_reward_points_be_used_by
         return can_reward_points_be_used_by(user)
     return user_passes_test(check_user)(view_func)
+
+
+# see https://mozilla-django-oidc.readthedocs.io/en/stable/
+class OpenIDAuthenticationBackend(OIDCAuthenticationBackend):
+    def filter_users_by_claims(self, claims):
+        email = claims.get('email')
+        if not email:
+            return self.UserModel.objects.none()
+
+        user_email = email
+        for original_domain, replaced_domain in settings.INSTITUTION_EMAIL_REPLACEMENTS:
+            if email.endswith(original_domain):
+                user_email = email[:-len(original_domain)] + replaced_domain
+                break
+
+        try:
+            return UserProfile.objects.filter(email=user_email)
+        except UserProfile.DoesNotExist:
```

Comment:
I think `filter` never throws that (`get` does)

Reply:
i don't know why i put a `filter` there, should be `get`

Reply:
now i know: the return is expected to be a list (or QuerySet). so i made it a one element list now.

Reply:
Strange: In the documentation [here](https://mozilla-django-oidc.readthedocs.io/en/stable/installation.html#connecting-oidc-user-identities-to-django-users) they show an example where the user object is returned. Also, wouldn't it make sense to then also do this at the other returns (`return []`)?

Reply:
Hmm, relevant code is here: https://github.com/mozilla/mozilla-django-oidc/blob/3c3c4a4e3b991c4113c5424716e79c001d5b88e9/mozilla_django_oidc/auth.py#L315
That definitely expects something list-like (even though it throws an error when there is more than one element in the list :D)
I guess I'll open an issue on mozilla-django-oidc about this. I'd still argue we should make all the returns consistent, so the others should return empty lists.

Reply:
i'm returning empty lists now